core/common/src/Buffer.kt

```diff
@@ -39,17 +39,26 @@ import kotlin.jvm.JvmField
  * does not affect buffer's state and [exhausted] only indicates that a buffer is empty.
  */
 public class Buffer : Source, Sink {
-    @JvmField
+    @PublishedApi
```

Comment:
just curious[2]: does it make any difference (performance/code size/anything else wise) annotating it with `JvmField`? :)
so something like this:
```
@JvmField
@JvmSynthetic
@PublishedApi
internal var head: Segment? = null
```
I mean, looks like this will still work as intended.

Reply:
I've run some benchmarks on a diff library and had to push [this](https://github.com/petertrr/kotlin-multiplatform-diff/pull/112/commits/69e32f65886a463e3ea91e9e595c131cba4bfbc3) commit to solve a pretty obvious performance issue. If the field is accessed frequently I think annotating with `@JvmField` is always the best option.

Reply:
Having a publicly exposed field ties our hands when it comes to the future evolution of segments (or any other API). 

Any decent runtime/compiler would inline trivial methods (getters and setters usually match that criteria), so having a pair of a getter & a setter is almost the same as having a public field from a performance point of view.

Reply:
But it's the same if a user uses static-like imports:

```kotlin
object MyObject {
  fun one() {}
  fun two() {}
}
```

```kotlin
import com.test.MyObject.one
import com.test.MyObject.two

fun example() {
  one()
  two()
}
```

So I wouldn't worry too much about it.